spacy/cli/_util.py

```diff
@@ -221,24 +221,45 @@ def validate_project_commands(config: Dict[str, Any]) -> None:
 
     config (Dict[str, Any]): The loaded config.
     """
+
+    def verify_workflow_step(workflow_name: str, step: str) -> None:
+        if step not in command_names:
+            msg.fail(
+                f"Unknown command specified in workflow '{workflow_name}': {step}",
+                f"Workflows can only refer to commands defined in the 'commands' "
+                f"section of the {PROJECT_FILE}.",
+                exits=1,
+            )
+
     command_names = [cmd["name"] for cmd in config.get("commands", [])]
     workflows = config.get("workflows", {})
     duplicates = set([cmd for cmd in command_names if command_names.count(cmd) > 1])
     if duplicates:
         err = f"Duplicate commands defined in {PROJECT_FILE}: {', '.join(duplicates)}"
         msg.fail(err, exits=1)
-    for workflow_name, workflow_steps in workflows.items():
+    for workflow_name, workflow_step_or_lists in workflows.items():
         if workflow_name in command_names:
             err = f"Can't use workflow name '{workflow_name}': name already exists as a command"
             msg.fail(err, exits=1)
-        for step in workflow_steps:
-            if step not in command_names:
-                msg.fail(
-                    f"Unknown command specified in workflow '{workflow_name}': {step}",
-                    f"Workflows can only refer to commands defined in the 'commands' "
-                    f"section of the {PROJECT_FILE}.",
-                    exits=1,
-                )
+        for step_or_list in workflow_step_or_lists:
+            if isinstance(step_or_list, str):
+                verify_workflow_step(workflow_name, step_or_list)
+            else:
+                workflow_list = cast(List[str], step_or_list)
```

Comment:
Before casting here, we want to add an assert to check the type just for robustness sake.